### PR TITLE
fix:restore links to local resources

### PR DESCRIPTION
**Pull request by PKEv**
Added support for links like "file://192.168.1.1/" and "smb://localnode/"

### CODE REVIEW

**Comment by zetok** [comment]
Could you use https://github.com/tux3/qTox/blob/master/CONTRIBUTING.md#commit-message-format ?

**Comment by vith** [comment]
It isn't necessary to escape `.` inside a regex character class, right?

**Comment by zetok** [comment]
0e604af54021b69dcc593aaf3bb1058ddc13bd82: doesn't seem to work:
no highlight

Also, `[\.]` doesn't seem to be needed, since it's already ~covered by `\S`

```
\S  Matches a non-whitespace character.
```

?

Aside from "links" not working, is `file://` really a thing? I.e. I know that on ~unix `file:///` is a ~standard – is `file://` specified somewhere?

**Comment by PKEv** [comment]
strange...
qtox_screenshot_2016-04-11 21-30-28 484

**Comment by PKEv** [comment]
aboute "file:///...."
I can't check and was somewhat surprised as the links I have now are like: "file://192.168.1.8/Public" 

but look: http://stackoverflow.com/questions/18246053/how-can-i-create-a-link-to-a-local-file-on-a-locally-run-web-page/18246357

examples of different processing links:
qtox_screenshot_2016-04-11 22-02-38 351

it seems to me that we should support and format with two slashes...

**Comment by zetok** [comment]
Hm, looks like I might have compiled/launched wrong version, after recompiling `smb://` seems to work:
smb works

**Comment by zetok** [comment]
@PKEv working:

``` diff
diff --git a/src/chatlog/chatmessage.cpp b/src/chatlog/chatmessage.cpp
index 35327fd..017c34e 100644
--- a/src/chatlog/chatmessage.cpp
+++ b/src/chatlog/chatmessage.cpp
@@ -249,7 +249,9 @@ QString ChatMessage::detectAnchors(const QString &str)
                 "(?:\\b)((www\\.)|(http[s]?|ftp)://)" // (protocol)://(printable - non-special character)
                 // http://ONEORMOREALHPA-DIGIT
                 "\\w+\\S+)" // any other character, lets domains and other
-                "|(?:\\b)((file:///)|(smb://))([\\S| |\\.]*)" //link to a local file, valid until the end of the line
+                // ↓ link to a file, or samba share
+                //   https://en.wikipedia.org/wiki/File_URI_scheme
+                "|(?:\\b)((file|smb)://)([\\S| ]*)"
                 "|(?:\\b)(tox:[a-zA-Z\\d]{76})" //link with full user address
                 "|(?:\\b)(mailto:\\S+@\\S+\\.\\S+)" //@mail link
                 "|(?:\\b)(tox:\\S+@\\S+)"); // starts with `tox` then : and only alpha-digits till the end
```

**Comment by PKEv** [comment]
tested:
qtox_screenshot_2016-04-11 23-37-02 004
qtox_screenshot_2016-04-11 23-36-54 816

File: src/chatlog/chatmessage.cpp [hunk 1]
```diff
@@ -249,7 +249,7 @@ QString ChatMessage::detectAnchors(const QString &str)
                 "(?:\\b)((www\\.)|(http[s]?|ftp)://)" // (protocol)://(printable - non-special character)
                 // http://ONEORMOREALHPA-DIGIT
                 "\\w+\\S+)" // any other character, lets domains and other
-                "|(?:\\b)(file:///)([\\S| ]*)" //link to a local file, valid until the end of the line
+                "|(?:\\b)((file|smb)://)([\\S| ]*)" //link to a local file, valid until the end of the line
```
Comment (zetok) [inline src/chatlog/chatmessage.cpp:252]: could you correct the comment?

I.e. with `file://` or `smb://` it's no longer about only local files.

File: src/chatlog/chatmessage.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*
+﻿/*
```
Comment (zetok) [inline src/chatlog/chatmessage.cpp]:
``` diff
diff --git a/src/chatlog/chatmessage.cpp b/src/chatlog/chatmessage.cpp
index c2a563b..2bb986c 100644
--- a/src/chatlog/chatmessage.cpp
+++ b/src/chatlog/chatmessage.cpp
@@ -1,4 +1,4 @@
-/*
+<U+FEFF>/
```

Could you fix that?

**Comment by zetok** [comment]
Also, the commit message is kinda off – i.e. it's not really a fix, but introducing a new feature, where non-local `file://`s and `smb://` shares are now supported, so it should be:

```
feat(chatform): add support for non-local file and samba share links

E.g. links like "file://192.168.1.1/" and "smb://localnode/" are now
supported. Before only links to local files were supported, e.g.
"file:///local/file.txt".
```

File: src/chatlog/chatmessage.cpp (hunk 2 above)
Comment (PKEv) [inline src/chatlog/chatmessage.cpp]: i don' understand

**Pull request merged**